infra/src/main/kotlin/fi/fta/geoviite/infra/split/SplitService.kt

```diff
@@ -89,14 +97,216 @@ class SplitService(
         trackNumberId: IntId<TrackLayoutTrackNumber>,
     ): PublishValidationError? {
         val splitIds = splitDao.fetchUnfinishedSplitsByTrackNumber(trackNumberId)
-        return if (splitIds.isNotEmpty())
-            PublishValidationError(
-                PublishValidationErrorType.ERROR,
-                "$VALIDATION_SPLIT.split-in-progress"
-            )
+        return if (splitIds.isNotEmpty()) PublishValidationError(
+            PublishValidationErrorType.ERROR,
+            "$VALIDATION_SPLIT.split-in-progress",
+        )
         else null
     }
 
-    fun locationTrackHasAnySplitsNotDone(locationTrackId: IntId<LocationTrack>): Boolean =
-        splitDao.fetchUnfinishedSplits().any { it.isPartOf(locationTrackId) && it.bulkTransferState != BulkTransferState.DONE }
+    fun locationTrackHasAnySplitsNotDone(locationTrackId: IntId<LocationTrack>): Boolean = splitDao
+        .fetchUnfinishedSplits()
+        .any { it.isPartOf(locationTrackId) && it.bulkTransferState != BulkTransferState.DONE }
+
+    @Transactional
+    fun split(request: SplitRequest): IntId<SplitSource> {
```

Comment:
:meat_on_bone:

Reply:
Sinänsä aika monioperaatioinen funkkari, vaikka onkin ymmärrettävä. `splitin()` kutsumalla saa vähän sitä kuvaa, että hei nyt mentiin, mutta täällä kuitenkin edelleen haetaan useista lähteistä arvoja, joista moni kutsu voi epäonnistua. (Testaaminenkin täytyy nyt tehdä IT-testinä, mutta ehkäpä se on muutenkin järkevämpää.)

Vertauskuvallisemminhan tämä on vähän enemmän `trySplit` tai `maybeSplit` kuin `doSplit`.

Reply:
Ajatuksellisesti kyllä, mutta niinhän melkein kaikki operaatiot on, eikä niitä silti nimetä maybe:ksi vain sen takia että joku voi mennä pieleen (aina voi). Kaavin jo tuolta kuitenkin varsinaisen raiteen leikkaamisen puhtaaksi funkkariksi joten sen erikoisuudet voi yksikkötestata. Mutta kun tehdään sitä splittiä jonka UI käskee (kokonaisoperaatio) niin on pakko tehdä uudelleenlinkitykset jne, ja sitä on nykymallilla vaikea saada puhtaaksi.

Tuon uudelleenlinkityksen voisi toki rempata niin että koko relinking olisi muistinvarainen juttu joka tuottaisi listan lopputulos-vaihteita/raiteita. Sitten koko logiikan saisi puhtaaksi ja tänne jäisi vain 1. hae 2. tee muokkausjutut 3. tallenna. Se olisi nätti malli ja 2-stepin jutut voisi testata kokonaan unit testeillä.